src/libopal.c

```diff
@@ -2852,8 +2852,41 @@ gen_obj (char *asm_fn, char *obj_fn)
   /// Assert object file name is not null
   assert(obj_fn);
 
-  // TODO
+  /// Create the NASM call with -g, -f and -o flags
+  logger(DEBUG, "Calling NASM to assemble object.");
+  char NASM_cmd[1024] = { 0 };
+  sprintf(NASM_cmd, "NASM -g -f elf64 -o %s %s", asm_fn, obj_fn);
+  sprintf(perror_msg, "NASM -g -f elf64 -o %s %s", asm_fn, obj_fn);
```

Comment:
You don't need this if it's identical to NASM_cmd. You could just use NASM_cmd with perror().